src/services/drive/delete-file.ts

```diff
@@ -106,3 +126,35 @@ export async function deleteObjectStorageFile(key: string) {
 		Key: key
 	}).promise();
 }
+
+async function findRelatedNotes(fileId: string) {
+	// NOTE: When running raw query, TypeORM converts field name to lowercase. Wrap in quotes to prevent conversion.
+	const relatedNotes = await Notes.createQueryBuilder('note').where(':id = ANY("fileIds")', { id: fileId }).getMany();
+	for (const relatedNote of relatedNotes) {
+		const user = await Users.findOne({ id: relatedNote.userId });
+		if (user)
+			relatedNote.user = user;
+	}
+	return relatedNotes;
```

Comment:
Seems replaceable using JOIN
```ts
	const query = Notes.createQueryBuilder('note')
		.where(':id = ANY("fileIds")', { id: fileId })
		.leftJoinAndSelect('note.user', 'user');

	return await query.getMany();
```

Reply:
Probably looks unfixed.
I think the whole `findRelatedNotes` function will be replaced by the above.